test/cypress/integration/components/filter-blog-posts.js

```diff
@@ -0,0 +1,305 @@
+  beforeEach( () => {
+    cy.visit( '/about-us/blog/' );
+  } );
+  it( 'Item name search', () => {
+    // When I enter "loan" in the item name input field
+    blog.filterItemName( 'loan' );
+    // And I click "Apply filters" button
+    blog.applyFilters();
+    // Then I should see only results with the word "loan" in the post title
+    blog.resultsContent().should( 'contain', 'loan' );
+    // And the page url should contain "title=loan"
+    cy.url().should( 'include', 'title=loan' );
+  } );
+  it( 'Select a single category', () => {
+    // When I select the first checkbox in the Category list
+    filter.checkCategoryName( 'At the CFPB' );
+    // And I click "Apply filters" button
+    blog.applyFilters();
+    // Then I should see only results in that category
+    blog.notification().should( 'be.visible' );
+    // And the page url should contain "categories=at-the-cfpb"
+    cy.url().should( 'include', 'categories=at-the-cfpb' );
+  } );
+  it( 'Select multiple categories', () => {
+    // When I select all five checkboxes in the Category list
+    filter.checkCategoryName( 'At the CFPB' );
+    filter.checkCategoryId( 'directors-notebook' );
+    filter.checkCategoryId( 'policy_compliance' );
+    filter.checkCategoryId( 'data-research-reports' );
+    filter.checkCategoryName( 'Info for consumers' );
+    // And I click "Apply filters" button
+    blog.applyFilters();
+    // Then I should see only results that are in at least one of the selected categories
+    blog.notification().should( 'be.visible' );
+    // And the page url should contain "categories=at-the-cfpb"
+    cy.url().should( 'include', 'categories=at-the-cfpb' );
+    // And the page url should contain "categories=directors-notebook"
+    cy.url().should( 'include', 'categories=directors-notebook' );
+    // And the page url should contain "categories=policy_compliance"
+    cy.url().should( 'include', 'categories=policy_compliance' );
+    // And the page url should contain "categories=data-research-reports"
+    cy.url().should( 'include', 'categories=data-research-reports' );
+    // And the page url should contain "categories=info-for-consumers"
+    cy.url().should( 'include', 'categories=info-for-consumers' );
+  } );
+  it( 'Date range to present', () => {
+    // When I enter "01/01/2020" in the From date entry field
+    blog.filterFromDate( '2020-01-01' );
+    // And I click "Apply filters" button
+    blog.applyFilters();
+    // Then I shouldn't see results dated 12/31/2019 or older
+    blog.resultsHeaderRight().should( 'not.contain', '2019' );
+    blog.notification().should( 'be.visible' );
+    // And I paginate to the last page of results
+    page.lastResults();
+    // Then I should see only results dated 01/01/2020 or later
+    blog.lastResultHeader().should( 'contain', '2020' );
+    // And the page url should contain "from_date=2020-01-01"
+    cy.url().should( 'include', 'from_date=2020-01-01' );
+  } );
```

Comment:
I think this approach is good. It doesn't seem like an increasing number of results will present a major problem.

A couple suggestions for trimming it up slightly:

```suggestion
  it( 'Date range to present', () => {
    // When I enter "01/01/2020" in the From date entry field
    blog.filterFromDate( '2020-01-01' );
    // And I click "Apply filters" button
    blog.applyFilters();
    // Then the page url should contain "from_date=2020-01-01"
    cy.url().should( 'include', 'from_date=2020-01-01' );
    // And when I paginate to the last page of results
    page.lastResults();
    // Then I should see only results dated 01/01/2020 or later
    blog.lastResultHeader().should( 'contain', '2020' );
  } );
```

Reply:
Done. I trimmed this test up as you suggested.